pylint/checkers/variables.py

```diff
@@ -2670,39 +2733,55 @@ def _check_unpacking(
         # Attempt to check unpacking is properly balanced
         values = self._nodes_to_unpack(inferred)
         details = _get_unpacking_extra_info(node, inferred)
+
         if values is not None:
             if len(targets) != len(values):
-                # Check if we have starred nodes.
-                if any(isinstance(target, nodes.Starred) for target in targets):
-                    return
-                self.add_message(
-                    "unbalanced-tuple-unpacking",
-                    node=node,
-                    args=(
-                        details,
-                        len(targets),
-                        "" if len(targets) == 1 else "s",
-                        len(values),
-                        "" if len(values) == 1 else "s",
-                    ),
+                self._report_unbalanced_unpacking(
+                    node, inferred, targets, values, details
```

Comment:
```suggestion
                    "unbalanced-tuple-unpacking", node, inferred, targets, values, details
```

Reply:
Your suggestion to pass the symbol as an arg would make sense except here we would still need to do
```
symbol = (
            "unbalanced-dict-unpacking"
            if isinstance(inferred, DICT_TYPES)
            else "unbalanced-tuple-unpacking"
        )
 self._report_unbalanced_unpacking(symbol....)
```

There is only one place we can call ` self._report_unbalanced_unpacking("unbalanced-dict-unpacking"...` since that code path only handles that msg, but in this code path we need to differentiate.

I would like to politely push back and leave the code as is. I think it's quite clean to have the symbol check in only one location.

Reply:
Hmm right I missed the fact that we handled two cases here.